gtsfm/frontend/detector_descriptor/sift.py

```diff
@@ -39,28 +39,21 @@ def detect_and_describe(self, image: Image) -> Tuple[Keypoints, np.ndarray]:
             Corr. descriptors, of shape (N, D) where D is the dimension of each descriptor.
         """
 
-        # conert to grayscale
+        # Convert to grayscale.
         gray_image = image_utils.rgb_to_gray_cv(image)
 
-        # Creating OpenCV object
+        # Create OpenCV object.
         opencv_obj = cv.SIFT_create()
 
-        # Run the opencv code
+        # Run the OpenCV code.
         cv_keypoints, descriptors = opencv_obj.detectAndCompute(gray_image.value_array, None)
 
-        # convert to GTSFM's keypoints
+        # Convert to GTSFM's keypoints.
         keypoints = feature_utils.cast_to_gtsfm_keypoints(cv_keypoints)
 
-        # sort the features and descriptors by the score
-        # (need to sort here as we need the sorting order for descriptors)
-        sort_idx = np.argsort(-keypoints.responses)[: self.max_keypoints]
-
-        keypoints = Keypoints(
-            coordinates=keypoints.coordinates[sort_idx],
```

Comment:
@ayushbaid do you remember why we weren't just using `Keypoints.get_top_k()` here?

https://github.com/borglab/gtsfm/blob/master/gtsfm/common/keypoints.py#L89

Reply:
Because we need the indices for the descriptors.